specification/deviceprovisioningservices/Deviceprovisioningservices.Management/back-compatible.tsp

```diff
@@ -0,0 +1,119 @@
+import "@azure-tools/typespec-client-generator-core";
+
+using Azure.ClientGenerator.Core;
+using Microsoft.Devices;
+
+@@clientName(CertificateResponses.createOrUpdate::parameters.resource,
+  "certificateDescription"
+);
+@@clientName(CertificateResponses.verifyCertificate::parameters.body,
+  "request"
+);
+
+@@clientName(ProvisioningServiceDescriptions.createOrUpdate::parameters.resource,
+  "iotDpsDescription"
+);
+@@clientName(ProvisioningServiceDescriptions.update::parameters.properties,
+  "ProvisioningServiceTags"
+);
+
+@@clientName(PrivateEndpointConnections.createOrUpdatePrivateEndpointConnection::parameters.resource,
+  "privateEndpointConnection"
+);
+
+@@clientName(IotDpsResourceOperationGroup.checkProvisioningServiceNameAvailability::parameters.body,
+  "arguments"
+);
+
+// Add client location decorators for all operations with @operationId
+// CertificateResponses interface operations
+@@clientLocation(CertificateResponses.get, "DpsCertificate", "Get");
+@@clientLocation(CertificateResponses.createOrUpdate,
+  "DpsCertificate",
+  "CreateOrUpdate"
+);
+@@clientLocation(CertificateResponses.delete, "DpsCertificate", "Delete");
+@@clientLocation(CertificateResponses.list, "DpsCertificate", "List");
+@@clientLocation(CertificateResponses.generateVerificationCode,
+  "DpsCertificate",
+  "GenerateVerificationCode"
+);
+@@clientLocation(CertificateResponses.verifyCertificate,
+  "DpsCertificate",
+  "VerifyCertificate"
+);
+
+// GroupIdInformations interface operations
+@@clientLocation(GroupIdInformations.getPrivateLinkResources,
+  "IotDpsResource",
+  "GetPrivateLinkResources"
+);
+@@clientLocation(GroupIdInformations.listPrivateLinkResources,
+  "IotDpsResource",
+  "ListPrivateLinkResources"
+);
```

Comment:
I'm feeling confused by this comment. Are you saying getPrivateLinkResources is part of the group id information interface? I can see one point of view saying it is, yet I thought normally group id info is just exposed as part of the private endpoint connection details

Reply:
Details here: https://github.com/Azure/typespec-azure/issues/2683